# Project 4

In this project, you will summarize and present your analysis from Projects 1-3.

### Intro: Write a problem Statement/ Specific Aim for this project

Answer: Determine if there is an association between graduate school admission and the prestige of a student's undergraduate school using data from the UCLA admissions data set.

### Dataset:  Write up a description of your data and any cleaning that was completed

Answer: Data contains the following variables: 

Variable | Description | Type of Variable
---| ---| ---
Admit| 0 = not admitted 1 = admitted | categorical
GRE | GRE score 200-800 | continuous 
GPA | GPA 0-4.0 | continuous
Prestige | 1= not prestigious 2 = low prestige 3= good prestige 4= high prestige | categorical

The data is normally distributed, GPA and GRE are skewed toward the upper ranges but there is no need to correct the distribution. 
Cleaning: 
1) The raw dataset contained about 400 observations, removed missing values, leaving a total of 397 observations. Created 
2) Split prestige into 3 binary fields called dummy variables 

### Demo: Provide a table that explains the data by admission status

In [39]:
import pandas as pd
import numpy as np

df_raw = pd.read_csv("./datasets/admissions.csv")
df = df_raw.dropna() 

print(pd.crosstab(df['prestige'], df['admit'], colnames=['admit']))

print(df.loc[df['admit']==1]['gre'].mean(),
df.loc[df['admit']==0]['gre'].mean(),
df.loc[df['admit']==1]['gpa'].mean(),
df.loc[df['admit']==0]['gpa'].mean())



admit      0   1
prestige        
1.0       28  33
2.0       95  53
3.0       93  28
4.0       55  12
618.5714285714286 573.5793357933579 3.4892063492063485 3.347158671586714


Mean (STD) or counts by admission status for each variable 

| Not Admitted | Admitted
---| ---|---
GPA | 3.49  | 3.35
GRE |573.6 | 618.6
Prestige 1 | 28 | 33
Prestige 2 | 95 | 53
Prestige 3 |93 | 28
Prestige 4 |55 | 12

### Methods: Write up the methods used in your analysis

Answer:

1) Describe() was used to looked at the number of observations and the standard deviation of various variables
2) Boxplot was used to view the min, max, median, 25%, 75%, outliers 
3) .hist was used to see the distribution of the data 
4) created correlation matrix to test for colinearity 
5) Utilized logit funciton in statsmodels to perform logistic regression to predict admit using gre, gpa and prestige. Set prestige_1 as reference variable to compare other dummy variables (prestige_2, prestige_3, prestige_4) against. Looked at the CI and odds ratio. 
6) Used cartesian function to recreate the dataset with every logical combination of inputs to see how the predicted probability of admission increases/decreases across different variables. 
7) Made predictions using the enumerated dataset 


### Results: Write up your results

Answer:
1) Standard deviation for GRE is much larger than for GPA because GRE has a much larger range/mean.
2) min, max, median, 25%, 75%, outliers
3) normal distribution, GPA and GRE are slightly skewed toward the upper ranges
4) colinearity is not a concern 
5) Inverse relationship between the probability of being admitted and undergraduate school rank. Example: Students who attended a tier 2 undergraduate school had 0.50 the odds of being admitted to graduate school compared to students who attend a tier 1 undergraduate school.
6 and 7) Given the same GPA and GRE scores, students who attended a tier 4 undergraduate school had a 37% probability of being admitted into grad school, while student who attended a tier 1 school had a 73% likelihood of being admitted into grad school.


### Visuals: Provide a table or visualization of these results

<img src='./images/download.png' height= 25% width= 25%>

### Discussion: Write up your discussion and future steps

Answer: 